packages/SwingSet/src/vats/timer/vat-timer.js

```diff
@@ -1,98 +1,854 @@
 // @ts-check
 
-import { Nat } from '@agoric/nat';
-import { assert, details as X } from '@agoric/assert';
+import { E } from '@endo/eventual-send';
 import { Far, passStyleOf } from '@endo/marshal';
-import { makeNotifierFromAsyncIterable } from '@agoric/notifier';
 import { makePromiseKit } from '@endo/promise-kit';
-
-import { makeTimedIterable } from './timed-iteration.js';
+import { Nat } from '@agoric/nat';
+import { assert } from '@agoric/assert';
+import {
+  provideKindHandle,
+  defineDurableKind,
+  defineDurableKindMulti,
+  makeScalarBigMapStore,
+  makeScalarBigWeakMapStore,
+} from '@agoric/vat-data';
+import { provide, makeLegacyWeakMap } from '@agoric/store';
 import { TimeMath } from './timeMath.js';
 
-export function buildRootObject(vatPowers) {
+// RAM usage: O(number of outstanding delay() promises) +
+// O(number of notifiers??). setWakeup() only consumes DB.
+
+/**
+ * // Handler is a Far object with .wake(time) used for callbacks
+ *
+ * @typedef Handler
+ * @property {(scheduled: Timestamp) => unknown} wake
+ * @typedef {unknown} CancelToken
```

Comment:
```suggestion
 *
 * @typedef {unknown} CancelToken
```

Please separate distinct type definitions by at least a newline.